lib/fog/dynect/models/dns/record.rb

```diff
@@ -25,7 +25,9 @@ def save(replace=false)
           }
           options.delete_if {|key, value| value.nil?}
 
-          if replace
+          refresh_id if replace && identity.nil?
+          if replace && !identity.nil?
```

Comment:
Actually, my original comment was not entirely correct.  The real reason the logic was separate is because after `refresh_id` is called, `identity` may no longer be nil.  That was the main point of the patch.  :-)